### PR TITLE
A WAG at selective imports.

### PR DESCRIPTION
**Pull request by SamAxe**
This is an attempt at finding the selective imports in
dmd/compiler.

There were issues with the "automatic" finding of
selective imports with the files below.  Several
of the errors result in recursive template errors, so
these files are not touched and is a short list for
digging deeper into issues caused by selective
imports in them.

./src/dmd/frontend.d
./src/dmd/toir.d
./src/dmd/func.d
./src/dmd/statement.d
./src/dmd/compiler.d
./src/dmd/aggregate.d
./src/dmd/mtype.d
./src/dmd/arraytypes.d
./src/dmd/dsymbol.d
./src/dmd/tocsym.d
./src/dmd/dtemplate.d
./src/dmd/root/stringtable.d
./src/dmd/root/array.d
./src/dmd/statementsem.d
./src/dmd/tokens.d
./src/dmd/scanmscoff.d
./src/dmd/backend/cgreg.d
./src/dmd/backend/cgxmm.d
./src/dmd/backend/gflow.d
./src/dmd/backend/dtype.d
./src/dmd/backend/elfobj.d
./src/dmd/backend/dwarfdbginf.d
./src/dmd/backend/gdag.d
./src/dmd/backend/gother.d
./src/dmd/backend/machobj.d
./src/dmd/arrayop.d
./src/dmd/declaration.d
./src/dmd/expression.d
./src/dmd/attrib.d
./src/dmd/entity.d

### CODE REVIEW

File: compiler/src/dmd/ast_node.d [hunk 1]
```diff
@@ -10,8 +10,8 @@
  */
 module dmd.ast_node;
 
-import dmd.root.rootobject : RootObject;
-import dmd.visitor : Visitor;
+import dmd.root.rootobject  : RootObject;
+import dmd.visitor  : Visitor;
```
Comment (thewilsonator) [inline compiler/src/dmd/ast_node.d:14]: unneeded whitespace changes

File: compiler/src/dmd/argtypes_x86.d [hunk 2]
```diff
@@ -14,12 +14,12 @@ module dmd.argtypes_x86;
 import core.stdc.stdio;
 import core.checkedint;
 
-import dmd.astenums;
-import dmd.declaration;
-import dmd.globals;
-import dmd.mtype;
-import dmd.target;
-import dmd.visitor;
+import dmd.astenums;   // More than 10 symbols, not making selective.
```
Comment (thewilsonator) [inline compiler/src/dmd/argtypes_x86.d:17]: I realise this is a draft, but these comments should not be in the final diff.

File: compiler/src/dmd/apply.d [hunk 3]
```diff
@@ -11,13 +11,13 @@
...
 module dmd.apply;
 
-import dmd.arraytypes;
-import dmd.dsymbol;
-import dmd.dsymbolsem;
-import dmd.dtemplate;
-import dmd.expression;
-import dmd.root.array;
-import dmd.visitor;
+// import dmd.arraytypes;
```
Comment (thewilsonator) [inline compiler/src/dmd/apply.d:14]: For completely unused imports, if you could split them out into a different PR that would be much appreciated. That PR will be likely to be accepted much faster, will be much smaller.

**Review by thewilsonator — COMMENTED** [review]
Thanks for the initiative, this will be much easier to review as a set of smaller PRs

**Comment by SamAxe** [comment]
Thanks for comments.  I wanted to see how badly the tests failed in the CI.  Looks like I need to set up a windows environment to have a chance there.
